### DISCUSSION TITLE
Discord Integration

### DISCUSSION THREAD

**Discussion by herpaderpaldent** [seatplus/seatplus#4] [Ideas]
I'd love to hear from you, what you expect from a discord integration? What should it be able to do?

**Comment by pafro-dev** [comment]
Discord Messages in channels and as dm
Usermanagement (rename on discord to main char name) 
Rolemanagement (changing roles based on squads and groups)
Creating and Editing channels and roles

Access to notifications from other packages to send discord notifications.
For example: 
Pinging moon extractions on discord
Sending DMs with a tax bill to each member
Sending DMs when some errors with authentification and linked chars happen
etc.

**Comment by mikeamcbrien** [comment]
Any progress on this request?

**Comment by herpaderpaldent** [comment]
Hey @mikeamcbrien 

Indeed i am _slowly_ coming to an end. I need to write more tests to be really confident to release the package but i can share with you a current screenshot:

<img width="1403" alt="image" src="https://github.com/seatplus/seatplus/assets/6583519/0807e074-5c73-44b8-b2ba-4a4fbea2cce2">
<img width="1403" alt="image" src="https://github.com/seatplus/seatplus/assets/6583519/92e6bd0c-7ff7-4eae-9883-07cf416cd9c9">


Curious what are your thoughts when you see it?

**Reply by mikeamcbrien** [reply]
Oooh this is a nice start! This would be HUGE for my corp and help me manage our Discord immensely. 

What features or broadcast/subscription settings would this include?

Thanks for the update.

**Reply by herpaderpaldent** [reply]
> broadcast/subscription settings

Are you are asking me what notifications are available for subscription?

Instead of me telling you what notifications you can sign up for, why don't you tell me what kind of notifications you'd like to receive?

**Comment by herpaderpaldent** [comment]
I just released the first version of the discord integration. It supports:

- 2 Notifications
- Nickname Sync
- SquadSync

if you have the chance: let me know what you think of it. I assume there will be a few bugs but i tried to eliminate as much as possible and reached 100% code coverage with tests.

**Comment by zaxlofful** [comment]
What about notifications for if a structure is attacked?

I also vote for the moon mining notifications!
